backend/src/apiserver/storage/pipeline_store.go

```diff
@@ -175,38 +221,103 @@ func (s *PipelineStore) DeletePipeline(id string) error {
 }
 
 func (s *PipelineStore) CreatePipeline(p *model.Pipeline) (*model.Pipeline, error) {
+	// Set up creation time for both pipeline and pipeline version.
 	newPipeline := *p
 	now := s.time.Now().Unix()
 	newPipeline.CreatedAtInSec = now
+	newPipeline.DefaultVersion.CreatedAtInSec = now
+
+	// Set up UUID for pipeline and pipeline version.
 	id, err := s.uuid.NewRandom()
 	if err != nil {
 		return nil, util.NewInternalServerError(err, "Failed to create a pipeline id.")
 	}
 	newPipeline.UUID = id.String()
+	newPipeline.DefaultVersion.PipelineId = id.String()
```

Comment:
could you split the code to store a new pipeline and a new version? not mingle them together

Reply:
They need to be in a single transaction anyway? A further note: in the final release, we wouldn't insert/create versions here at all. CreatePipeline will only handle pipeline (as a container/bag) and has no insertion/creation of versions when everything is in its final status.